### PR TITLE
Develop to Master

### PR DESCRIPTION
**Pull request by krupalmodi18**
- Python 3 upgrade
- Newman test runner

### CODE REVIEW

File: postman_tests/lib/datastore.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    """
+    for file_path in glob.glob(os.path.join(datastore_data_path, '*.csv')):
+        entity_name = common.get_entity_name(file_path)
+        print(f"Syncing {entity_name}, mode: {mode}")
+        contents = convert_csv_to_dict(file_path, mode)
+        url = get_api_url(config_file_path)
+        try:
+            req = requests.post(f"{url}/{entity_name}", data=json.dumps(contents))
+            req.raise_for_status()
+        except requests.exceptions.HTTPError as e:
```
Comment (srajdev) [inline postman_tests/lib/datastore.py:34]: F841 local variable 'e' is assigned to but never used

File: postman_tests/lib/newman.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
+from __future__ import absolute_import
+import json
+import os
+import glob
+import itertools
+from . import common
+
+
+def check_if_data_valid(entities_data_path):
+    # type: (str) ->  boolean
```
Comment (srajdev) [inline postman_tests/lib/newman.py:10]: F821 undefined name 'boolean'

File: postman_tests/lib/newman.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                    return False
+        return True
+    except Exception as e:
+        print('Invalid json data in ' + path
+              + '\n' + str(e))
+        raise e
+
+
+def read_entities_data(entities_data_path):
+    # type: (str) -> Dict[str, List[Dict[str, Any]]]
```
Comment (srajdev) [inline postman_tests/lib/newman.py:42]: F821 undefined name 'Dict'
F821 undefined name 'List'
F821 undefined name 'Any'

File: postman_tests/lib/newman.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                for key, _ in d['input'].items():
+                    d1[entity + '_' + key] = d['input'][key]
+                d1[entity + '_expected'] = d['expected']
+                modified_data.append(d1)
+            entities_data[entity] = modified_data
+    return entities_data
+
+
+def generate_newman_data(entities_data_path):
+    # type: (str) -> List[Dict[str, Any]]
```
Comment (srajdev) [inline postman_tests/lib/newman.py:70]: F821 undefined name 'List'
F821 undefined name 'Dict'
F821 undefined name 'Any'

File: datastore_setup.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import os
```
Comment (srajdev) [inline datastore_setup.py:1]: W1618 [34mimport missing `from __future__ import absolute_import`[0m

File: datastore_setup.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+BASE_DIR = os.path.dirname(__file__)
+
+
+# Below needs to be committed if you want to use existing data in the Elasticsearch Setup
+# TODO move this part to a different script and run on-demand
+# Comment out entire section if you want to reuse existing data
+
+def setup_datastore():
+    db = DataStore()
+    print("Setting up DataStore for Chatbot NER")
```
Comment (srajdev) [inline datastore_setup.py:15]: E1601 [1;31mprint statement used[0m

File: datastore_setup.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    print("Deleting any stale data ...")
```
Comment (srajdev) [inline datastore_setup.py:16]: E1601 [1;31mprint statement used[0m

File: datastore_setup.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    db.delete(err_if_does_not_exist=False)
+    print("Creating the structure ...")
```
Comment (srajdev) [inline datastore_setup.py:18]: E1601 [1;31mprint statement used[0m

File: datastore_setup.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    db.create(err_if_exists=True)
+    print("Populating data from " + os.path.join(BASE_DIR, 'data', 'entity_data') + " ...")
```
Comment (srajdev) [inline datastore_setup.py:20]: E1601 [1;31mprint statement used[0m

File: datastore_setup.py [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    db.populate(entity_data_directory_path=DEFAULT_ENTITY_DATA_DIRECTORY)
+    print("Done!")
```
Comment (srajdev) [inline datastore_setup.py:22]: E1601 [1;31mprint statement used[0m

File: postman_tests/lib/newman.py [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+    """
+    try:
+        path = ''
+        for file_path in glob.glob(os.path.join(entities_data_path, '*.json')):
+            path = file_path
+            with open(file_path, 'r') as file:
+                data = json.load(file)
+            for item in data:
+                if not all(k in item for k in ('input', 'expected')):
+                    print('Invalid json data in ' + file_path)
```
Comment (srajdev) [inline postman_tests/lib/newman.py:32]: E1601 [1;31mprint statement used[0m

File: postman_tests/lib/newman.py [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+                    return False
+        return True
+    except Exception as e:
+        print('Invalid json data in ' + path
```
Comment (srajdev) [inline postman_tests/lib/newman.py:36]: E1601 [1;31mprint statement used[0m

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Pull request merged**